engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/immutable/AtVectorNode.java

```diff
@@ -0,0 +1,41 @@
+package org.enso.interpreter.node.expression.builtin.immutable;
+
+import com.oracle.truffle.api.CompilerDirectives;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.interop.InvalidArrayIndexException;
+import com.oracle.truffle.api.interop.UnsupportedMessageException;
+import com.oracle.truffle.api.nodes.Node;
+import org.enso.interpreter.dsl.BuiltinMethod;
+import org.enso.interpreter.node.expression.builtin.interop.syntax.HostValueToEnsoNode;
+import org.enso.interpreter.runtime.Context;
+import org.enso.interpreter.runtime.data.Vector;
+import org.enso.interpreter.runtime.error.DataflowError;
+import org.enso.interpreter.runtime.error.PanicException;
+
+@BuiltinMethod(
+    type = "Vector",
+    name = "at",
+    description = "Returns an element of Vector at the specified index.")
+public class AtVectorNode extends Node {
+  private @Child InteropLibrary interop = InteropLibrary.getFactory().createDispatched(3);
+  private @Child HostValueToEnsoNode toEnso = HostValueToEnsoNode.build();
+
+  Object execute(Vector self, long index) {
+    try {
+      long actualIndex = index < 0 ? index + self.length(interop) : index;
+      return self.readArrayElement(actualIndex, interop, toEnso);
+    } catch (InvalidArrayIndexException e) {
+      Context ctx = Context.get(this);
+      try {
+        return DataflowError.withoutTrace(
+            ctx.getBuiltins().error().makeIndexOutOfBoundsError(index, self.length(interop)), this);
+      } catch (UnsupportedMessageException ex) {
+        CompilerDirectives.transferToInterpreter();
+        throw new PanicException(e.getMessage(), this);
+      }
+    } catch (UnsupportedMessageException e) {
+      CompilerDirectives.transferToInterpreter();
+      throw new PanicException(e.getMessage(), this);
+    }
```

Comment:
Cannot this be a single common `catch`?

Reply:
Yes, but if we moved this catch to be 'outside' wouldn't we catch also that exception thrown from handler?

Reply:
I refactored it a bit. We will still end up with 2 try blocks though.

Reply:
Yes, but now we don't have two almost exactly the same catch branches. Personally, I like the new version much better :) Thanks!